src/redux/user/userReducer.js

```diff
@@ -0,0 +1,55 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { toast } from 'react-toastify';
+import jwt_decode from 'jwt-decode';
+import Parse from 'parse';
+
+Parse.initialize(process.env.REACT_APP_PARSE_APP_ID);
+Parse.serverURL = process.env.REACT_APP_SERVER_URL;
+
+export const initializeGoogleLoginThunk = createAsyncThunk('user/initializeGoogleLogin', async (_, { dispatch }) => {
+    google.accounts.id.initialize({
+        client_id: process.env.REACT_APP_CLIENT_ID,
+        callback: (response) => {
+            const userObject = jwt_decode(response.credential);
+            dispatch(setUser(userObject));
+            Parse.User.logInWith('google', { authData: { id: userObject.sub, access_token: response.credential } })
+                .catch(error => {
+                    toast.error('Error logging in through Google.');
+                });
+        }
+    });
+
+    google.accounts.id.renderButton(
+        document.getElementById("signInDiv"),
+        { theme: "outline", size: "large" }
+    );
+
+    google.accounts.id.prompt();
+});
+
+export const logoutThunk = createAsyncThunk('user/logout', async (_, { dispatch }) => {
+    await Parse.User.logOut();
+    dispatch(resetUser());
+});
+
+const userSlice = createSlice({
+  name: 'user',
+  initialState: {},
```

Comment:
initialState: {auth: {}},

We may have other attributes about the user in the future that's independent from login info. So we need to store login details inside auth, not the entire state